### PR TITLE
storage: remove kv.MemCache in favor of Hashicorp's LRU

### PR DESCRIPTION
**Pull request by brendoncarroll**
Pulled out of #8643.  Before Go had generics we were serializing everything and storing it in caches as bytes.  We no longer need to do that.

### CODE REVIEW

File: src/internal/obj/minio_client.go [hunk 1]
```diff
@@ -48,8 +48,9 @@ func newMinioClientV2(endpoint, bucket, id, secret string, secure bool) (*minioC
 func (c *minioClient) Put(ctx context.Context, name string, r io.Reader) (retErr error) {
 	defer func() { retErr = c.transformError(retErr, name) }()
 	opts := minio.PutObjectOptions{
-		ContentType: "application/octet-stream",
-		PartSize:    uint64(8 * 1024 * 1024),
+		ContentType:      "application/octet-stream",
+		PartSize:         uint64(8 * 1024 * 1024),
+		DisableMultipart: true,
```
Comment (brendoncarroll) [inline src/internal/obj/minio_client.go:53]: Multipart uploads are generally bad for performance when dealing with our typical chunk sizes.  We use Go CDK for importing and exporting large files anyways.  But this is used in tests.

File: src/internal/storage/chunk/transform.go [hunk 2]
```diff
@@ -45,11 +44,13 @@ func Create(ctx context.Context, opts CreateOptions, ptext []byte, createFunc fu
 
 // Get calls client.Get to retrieve a chunk, then verifies, decrypts, and decompresses the data.
 // cb is called with the uncompressed plaintext
```
Comment (jrockway) [inline src/internal/storage/chunk/transform.go:46]: cb is no longer an arg

File: src/internal/storage/fileset/postgres_store.go [hunk 3]
```diff
@@ -15,23 +16,26 @@ import (
...
 	db      *pachsql.DB
-	cache   kv.GetPut
+	cache   *lru.Cache[ID, *Metadata]
 	deduper *miscutil.WorkDeduper[ID]
 }
 
 // NewPostgresStore returns a Store backed by db
 // TODO: Expose configuration for cache size?
 func NewPostgresStore(db *pachsql.DB) MetadataStore {
+	mc, err := lru.New[ID, *Metadata](100)
```
Comment (jrockway) [inline src/internal/storage/fileset/postgres_store.go:32]: I think the configuration option StorageMemoryCacheSize needs to go away, or be read here.

Comment (brendoncarroll) [reply src/internal/storage/fileset/postgres_store.go:32]: I believe StorageMemoryCacheSize refers to the size of the chunk cache.  We don't expose a way to configure this.

Comment (jrockway) [reply src/internal/storage/fileset/postgres_store.go:32]: You're deleting some code that reads it here, though: https://github.com/pachyderm/pachyderm/pull/8952/files#diff-f20d955c17bf7f6b9947ddec0131271bce16608c1f09b2d5dfb18092a66d37daL50

Comment (jrockway) [reply src/internal/storage/fileset/postgres_store.go:32]: Nevermind, I see your fix now :)

**Review by jrockway — APPROVED** [review]
LGTM.

File: src/internal/storage/fileset/postgres_store.go (hunk 3 above)
Comment (brendoncarroll) [reply src/internal/storage/fileset/postgres_store.go:32]: Yeah good catch.  I just added it back.  We configure that in `storage/config.go` now.

**Pull request merged**